chore: mantine-fy dashboard alert

joaoviana opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes: N/A

### Description:

before
<img width="815" alt="Screenshot 2023-12-14 at 22 14 04" src="https://github.com/lightdash/lightdash/assets/7611706/34839a97-44aa-4311-89ce-ca39f5077aee">

after
<img width="1296" alt="Screenshot 2023-12-14 at 22 20 06" src="https://github.com/lightdash/lightdash/assets/7611706/644073a3-ce5c-463e-b3b7-1012924e868d">

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

IrakliJani commented:
@joaoviana 

ui nits:

we should probably change the layout of this:

![CleanShot 2023-12-15 at 19 31 48@2x](https://github.com/lightdash/lightdash/assets/962095/9759c393-83bf-4531-9673-29b20500d734)

option one: move icon to the left and text align left
option two: center action icons as well

IrakliJani commented on packages/frontend/src/pages/Dashboard.tsx:
```diff
@@ -501,24 +500,39 @@ const Dashboard: FC = () => {
+                        </Text>
+                    </Group>
+                    <Group position="right">
+                        <Button>Stay</Button>
```
does stay close the modal?

joaoviana replied:
🤦

joaoviana commented:
@IrakliJani 

<img width="955" alt="Screenshot 2023-12-15 at 16 09 17" src="https://github.com/lightdash/lightdash/assets/7611706/7dfe0ea8-581c-4086-af13-117840a4c169">

fixed the issue with `stay` not closing the modal
changed layout ( a mix of your suggestions)
user can't close on click outside

IrakliJani commented:
merging
